pkg/vpcmodel/connectivityAbstraction.go

```diff
@@ -152,8 +120,66 @@ func mergeConnectivityWithNodeSetAbstraction(
 		for _, conns := range nodeConns {
 			allConns = allConns.Union(conns)
 		}
-		res.updateAllowedConnsMap(src, nodeSet, allConns)
+		mergedConnectivity.updateAllowedConnsMap(src, nodeSet, allConns)
 	}
+	return mergedConnectivity
+}
 
-	return res
+// nodeSetAbstractionInformation() collects abstraction information of the nodeSet.
+// for now, it collects the connections that are missing for full abstraction.
+func (nsa *NodeSetAbstraction) nodeSetAbstractionInformation(mergedConnectivity GeneralConnectivityMap,
+	nodeSetToNodeSet, otherFromNodeSet, otherToNodeSet GeneralConnectivityMap,
+	nodeSet NodeSet) *AbstractionInfo {
+	abstractionInfo := &AbstractionInfo{}
+	abstractionInfo.missingEgressConnections = nsa.missingConnections(otherFromNodeSet, mergedConnectivity, nodeSet, false)
+	abstractionInfo.missingIngressConnections = nsa.missingConnections(otherToNodeSet, mergedConnectivity, nodeSet, true)
+	abstractionInfo.missingIngressConnections.addMap(nsa.missingConnections(nodeSetToNodeSet, mergedConnectivity, nodeSet, true))
```

Comment:
Why is `nodeSetToNodeSet` considered `ingress`?
If it is arbitrary and could have been egress - please mention it and explain.
If it is ingress for a reason  - add an explanation as well

Reply:
done

Reply:
fixed

Reply:
fixed spelling mistakes and pushed